packages/web/src/components/gcds-input/gcds-input.tsx

```diff
@@ -318,14 +507,33 @@ export class GcdsInput {
 
     this.validateValidator();
 
-    this.inheritedAttributes = inheritAttributes(this.el, this.shadowElement, [
-      'placeholder',
-    ]);
+    this.inheritedAttributes = inheritAttributes(this.el, this.shadowElement);
 
     this.internals.setFormValue(this.value ? this.value : null);
     this.initialValue = this.value ? this.value : null;
   }
 
+  componentDidLoad() {
+    let lengthValidity;
+    // maxlength/minlength validation on load
+    if (this.value && (this.minlength || this.maxlength)) {
+      if (this.minlength && this.value.length < this.minlength) {
+        lengthValidity = { tooShort: true };
+      } else if (this.maxlength && this.value.length > this.minlength) {
```

Comment:
Is `this.minlength` supposed to be `this.maxlength`? `tooLong: true` looks like we want to check for a max length here.